server/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -553,11 +554,15 @@ private static String toString(Collection<String> items) {
      * @param shardId the id of the shard to delete to delete
      * @throws IOException if an IOException occurs
      */
-    public void deleteShardDirectorySafe(ShardId shardId, IndexSettings indexSettings) throws IOException, ShardLockObtainFailedException {
+    public void deleteShardDirectorySafe(
+        ShardId shardId,
+        IndexSettings indexSettings,
+        Consumer<List<Path>> listener
```

Comment:
All callers call `List#of(Path[])` on the listener, maybe we should make this a `Consumer<Path[]>` instead?

Reply:
Ok, I pushed 850d1a389665e4d644e6c730691e0f2debd26e41 to use `Path[]` all over the place